packages/eslint-plugin-react/rules/throttleUseResizeObserverCallback.ts

```diff
@@ -0,0 +1,101 @@
+import { type Rule } from 'eslint';
+
+const messages = {
+  missingThrottleFunction: 'Callback function in useResizeObserver should be throttled.',
+  wrapWithUseRafCallback: 'Wrap callback with useRafCallback',
+  wrapWithUseDebounceCallback: 'Wrap callback with useDebounceCallback',
+  wrapWithUseThrottleCallback: 'Wrap callback with useThrottleCallback',
+};
+
+const defaultThrottleFunctionNames = [
+  'useRafCallback',
+  'useDebounceCallback',
+  'useThrottleCallback',
+  'throttle',
+  'debounce',
+  'raff',
+];
+
+export default {
+  meta: {
+    messages,
+    docs: {
+      description: 'Callback function in useResizeObserver should be throttled.',
+      recommended: true,
+      url: 'https://example.com/',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          throttleFunctionNames: {
+            description: 'The throttle function identifiers',
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    ],
+    hasSuggestions: true,
+  },
+
+  create(context): Rule.RuleListener {
+    return {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      CallExpression(node): void {
+        if ('name' in node.callee && node.callee.name !== 'useResizeObserver') {
```

Comment:
We should make useResizeObserver configurable as well so that we can support for the `useMutationObserver` hook.